extensions/vscode/src/utils/config.ts

```diff
@@ -0,0 +1,47 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+function getConfigFile(configFilePath: string): string | null {
+  const configDirs = process.env.XDG_CONFIG_DIRS ? 
+         process.env.XDG_CONFIG_DIRS.split(':') : 
+         ['/etc'];
+
+  // Iterate over the directories and check if the file exists
+  for (const dir of configDirs) {
+    const filePath = path.join(dir, configFilePath);
+    if (fs.existsSync(filePath)) {
+      return filePath;
+    }
+  }
+
+  return null;
+}
+
+
+export function getXDGConfigProperty(configFilePath: string, propertyName: string): string | null {
+  console.log("In getXDGConfigProperty with: " + configFilePath + " and: " + propertyName);
+  let filePath = getConfigFile(configFilePath);
+  if (!filePath) {
+    console.log("No file path with path found");
+    return null;
+  }
+  const fileContent = fs.readFileSync(filePath, 'utf-8');
```

Comment:
It may be optimal to use `fs.readFile` and `fs.exists` above and make both of these functions asynchronous then `await` their calls in the two `multiStepInputs` files.

That would prevent other VS Code operations from halting while files are being read or checked.

Reply:
Yeah, I thought about doing that but it makes it awkward in the code for config values to be async. Since it's now linux only, and will just be an extra 'stat' call to a local directory for most users maybe it's ok? Or I could load the value when the plugin starts. Happy to do that if you think it's worth it.

Reply:
I agree with @dotNomad's opinion, so if you don't mind, yes please.